plugins/modules/database/postgresql/postgresql_set.py

```diff
@@ -388,8 +385,8 @@ def main():
         kw['restart_required'] = restart_required
         module.exit_json(**kw)
 
-    # Set param:
-    if value and value != current_value:
+    # Set param (value can be an empty string):
+    if (value or value == '') and value != current_value:
```

Comment:
I’m not familiar with this codebase, but since value is of type string, how could it be falsey without being an empty string?

```suggestion
    if value != current_value:
```

Reply:
value can be `None`

Reply:
if it isn't passed explicitly

Reply:
e.g. in case of `reset` passed, value will be `None`

Reply:
`value or value == ''` would be equivalent to `value is not None` then.

Reply:
True and False are also not good

Reply:
I don't think `True` and `False` are valid values here. The module option is defined as optional `str`, and the code before does not seem to change it to anything else than `str`.

Reply:
i think your suggestion should work:) thanks!